sql: go 1.12 makes tests fail in sqlbase

Running `make test PKG=./pkg/sql/sqlbase` with go 1.12 on my machine:

```
--- FAIL: TestEncDatumSize (0.00s)
    encoded_datum_test.go:656: on {0 [] 1.23E+4}        received 104, expected 136
    encoded_datum_test.go:656: on {0 [44 3 46 0] 1.23E+4}       received 108, expected 140
    encoded_datum_test.go:668: on [{0 [136] <nil>} {1 [135 132] <nil>} {0 [247 48 57] <nil>} {0 [] 123} {0 [246 123] 123} {0 [4] <nil>} {1 [3 63 161 63 255 255 255 255 255] <nil>} {0 [] 123.0} {0 [5 64 94 192 0 0 0 0 0] 123.0} {0 [39] <nil>} {1 [34 252 209 0] <nil>} {0 [] 1.23E+4} {0 [44 3 46 0] 1.23E+4} {0 [18 0 1] <nil>} {1 [19 206 205 204
 29 115 103 255 254] <nil>} {0 [] '12'} {0 [18 49 50 51 52 0 1] '12345'} {0 [20 249 91 50 40 104 136] <nil>} {0 [20 249 91 50 40 116 248 52 191 21] <nil>}]     received 1191, expected 1255

```

--- a/pkg/sql/sqlbase/encoded_datum_test.go
+++ b/pkg/sql/sqlbase/encoded_datum_test.go
@@ -546,11 +546,13 @@ func TestEncDatumSize(t *testing.T) {
 		asc  = DatumEncoding_ASCENDING_KEY
 		desc = DatumEncoding_DESCENDING_KEY
 
-		DIntSize           = unsafe.Sizeof(tree.DInt(0))
-		DFloatSize         = unsafe.Sizeof(tree.DFloat(0))
-		DDecimalStructSize = unsafe.Sizeof(tree.DDecimal{Decimal: *apd.New(0, 0)})
-		DStringSize        = unsafe.Sizeof(*tree.NewDString(""))
+		DIntSize    = unsafe.Sizeof(tree.DInt(0))
+		DFloatSize  = unsafe.Sizeof(tree.DFloat(0))
+		DStringSize = unsafe.Sizeof(*tree.NewDString(""))
 	)
+
+	dec12300 := &tree.DDecimal{Decimal: *apd.New(123, 2)}
+	decimalSize := dec12300.Size()
 
 	testCases := []struct {
 		encDatum     EncDatum
@@ -609,16 +611,16 @@ func TestEncDatumSize(t *testing.T) {
 			expectedSize: EncDatumOverhead + 4, // 123.0 is encoded with length 4 byte array
 		},
 		{
-			encDatum:     DatumToEncDatum(ColumnType{SemanticType: ColumnType_DECIMAL}, &tree.DDecimal{Decimal: *apd.New(123, 2)}),
-			expectedSize: EncDatumOverhead + DDecimalStructSize + 40, // In this case, DDecimal references 40 bytes for internal things
+			encDatum:     DatumToEncDatum(ColumnType{SemanticType: ColumnType_DECIMAL}, dec12300),
+			expectedSize: EncDatumOverhead + decimalSize,
 		},
 		{
 			encDatum: EncDatum{
 				encoding: asc,
-				encoded:  encoding.EncodeDecimalAscending(nil, apd.New(123, 2)),
-				Datum:    &tree.DDecimal{Decimal: *apd.New(123, 2)},
+				encoded:  encoding.EncodeDecimalAscending(nil, &dec12300.Decimal),
+				Datum:    dec12300,
 			},
-			expectedSize: EncDatumOverhead + 4 + DDecimalStructSize + 40, // 12300.0 is encoded with length 4 byte array
+			expectedSize: EncDatumOverhead + 4 + decimalSize,
 		},
 		{
 			encDatum:     EncDatumFromEncoded(asc, encoding.EncodeStringAscending(nil, "")),

